### DISCUSSION TITLE
Problem when calculating variant width

### DISCUSSION THREAD

**Discussion by jlhostis** [ORTools4MTEX/ORTools#18] [Q&A] [answered]
Hello ORTools team, thank you very much for your very complete add-on and suite of tutorials, they are really helpful.

I am having an issue following example7 when using one of my EBSD maps and calculating `plotMap_blockWidths` I get the following error : 
```
Array indices must be positive integers or logical values.

Error in indexing (line 7)
    v.x = subsref(v.x,s(1));

Error in plotMap_blockWidths>calcBlockWidth (line 145)
hh = hh(job.packetId(job.mergeId == pGrainId))';

Error in plotMap_blockWidths (line 48)
        [dBlockNew{ii},~,~] = calcBlockWidth(job,pGrain,pGrainId(ii),cGrains{ii});

Error in PAGjules_v03 (line 197)
plotMap_blockWidths(job,'linewidth',1);
```
All the other functions are working fine, with the example map or with my maps. I suspected the issue was that I have 3 phases indexed (martensite, austenite and Fe3C), but even when only selecting `ebsd('Martensite','Austenite','notIndexed')`  it's still not working.

I'm including my *import data* section for reference, maybe some mistake is made here :
```matlab
%% Load data
% Crystal symmetry w/ Fe3C
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Iron bcc (old)', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('m-3m', [3.7 3.7 3.7], 'mineral', 'Iron fcc', 'color', [0.56 0.74 0.56]),...
  crystalSymmetry('mmm', [5.1 6.8 4.5], 'mineral', 'Fe3C', 'color', [0.85 0.65 0.13])};
% path to files
% pname = '/home/lhostis/MATLAB-ebsd/ORTools-2.0.0';
pname = 'E:\Documents\MATLAB\ORTools-2.0.0';
% which files to be imported
fname = [pname '/EBSD Map Data 6 - EBSD Data Map_001.ctf'];
% create an EBSD variable containing the data
ebsd_bckp = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame')
```

The script and the data are also linked here :
[PAGjules_and_data.zip](https://github.com/ORTools4MTEX/ORTools/files/9784086/PAGjules_and_data.zip)

**Comment by frankNiessen** [comment, marked as answer]
Dear Jules, 

I have tried to track your issue down and it is related to our limited focus on microstructures that have retained austenite, i.e. austenite that is not reconstructed from martensite. I have now implemented some bugfixes to make things work for you, you should download the latest [ORTools developer release](https://github.com/ORTools4MTEX/ORTools/archive/develop.zip). 

That being said, I do not think that your reconstruction looks very reliable. If you check the orientations of your retained austenite and compare with the reconstructed austenite, they often don't match.  You also only have characterized parts of your austenite grains, for a good reconstruction it would be good to at least span a handful of grains.

**Reply by jlhostis** [reply]
Hi Franck,

thanks for your quick response. I downloaded the latest developer release and the map is plotting, but only on a few blocks (which are not correlated with the poorly fitted PAGs excluded during cleaning phase).

I could not manage to compute the block width on all blocks, even when reproducing every step of example 7, as you can see on the picture below in a bigger map :
  
![blocwidth](https://user-images.githubusercontent.com/41578866/196518416-a072874e-212e-4290-b91d-c67185a31a91.jpg)

You will find my dataset and script enclosed here :
[BlocWidth_jules.zip](https://github.com/ORTools4MTEX/ORTools/files/9814212/BlocWidth_jules.zip)

Indeed, I agree with your remark on my map's size, I am planning on recording bigger EBSD maps to cover several austenite grains.

Best regards, 
Jules

**Reply by frankNiessen** [reply]
Hi Jules, 

sorry for the long waiting time, this issue should now be fixed in [release 2.0.2](https://github.com/ORTools4MTEX/ORTools/releases/tag/v2.0.2).

Let me know if it works for you. 

Best wishes
Frank

**Reply by jlhostis** [reply]
Hello Franck, 

Thank you very much for correcting the bugs, everything seems to be working perfectly fine now.

Sincerely,
Jules
